core/src/main/java/me/mykindos/betterpvp/core/client/offlinemessages/OfflineMessage.java

```diff
@@ -0,0 +1,91 @@
+package me.mykindos.betterpvp.core.client.offlinemessages;
+
+import lombok.Data;
+import lombok.Getter;
+import me.mykindos.betterpvp.core.inventory.item.ItemProvider;
+import me.mykindos.betterpvp.core.utilities.UtilMessage;
+import me.mykindos.betterpvp.core.utilities.UtilTime;
+import me.mykindos.betterpvp.core.utilities.model.description.Describable;
+import me.mykindos.betterpvp.core.utilities.model.description.Description;
+import me.mykindos.betterpvp.core.utilities.model.item.ItemView;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import net.kyori.adventure.text.minimessage.MiniMessage;
+import net.kyori.adventure.text.minimessage.tag.resolver.TagResolver;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.entity.Player;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+@Data
+public class OfflineMessage implements Describable {
+    private final UUID client;
+    private final long time;
+    private final Action action;
+    private final String rawContent;
+
+
+    public Component getContent() {
+        return Component.empty().color(NamedTextColor.WHITE)
+                .append(MiniMessage.miniMessage().deserialize(rawContent, TagResolver.standard()));
+    }
+
+    public void send() {
+        Player player = Bukkit.getPlayer(client);
+        if (player != null) {
+            Component message = Component.text(UtilTime.getDateTime(time)).appendSpace().append(getContent());
+            UtilMessage.message(player, action.getTitle(), message);
+        }
+
+    }
+
+    /**
```

Comment:
if you're desperate to keep long as the timestamp field, add an instant method and use this to fetch how long ago something was, will be very useful for displaying 'x ago'